File: leetcode/solutions/0235.LowestCommonAncestorOfABinarySearchTree/lowestCommonAncestor_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
	ds "github.com/yukisato/leetcode-util/datastructure"
)

func Test_0235_LowestCommonAncestorOfABinarySearchTree(t *testing.T) {
	{
		arg1 := ds.CreateTreeNodeFromString([]string{"6", "2", "8", "0", "4", "7", "9", "null", "null", "3", "5"})
		arg2 := arg1.Left
		arg3 := arg1.Right
		got := lowestCommonAncestor(arg1, arg2, arg3)

		if !cmp.Equal(got, arg1) {
			t.Errorf("got %#v want %#v", got, arg1)
		}
	}

	{
		arg1 := ds.CreateTreeNodeFromString([]string{"6", "2", "8", "0", "4", "7", "9", "null", "null", "3", "5"})
		arg2 := arg1.Left
		arg3 := arg1.Left.Right
		got := lowestCommonAncestor(arg1, arg2, arg3)

		if !cmp.Equal(got, arg2) {
			t.Errorf("got %#v want %#v", got, arg2)
		}
	}
}

File: leetcode/solutions/0235.LowestCommonAncestorOfABinarySearchTree/lowestCommonAncestor_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import lowestCommonAncestor from "./lowestCommonAncestor.ts";
import { createBinaryTreeNode } from "../../../datastructure/BinaryTreeNode.ts";

Deno.test("0235. Lowest Common Ancestor of a Binary Search Tree", () => {
  {
    const arg1 = createBinaryTreeNode([6, 2, 8, 0, 4, 7, 9, null, null, 3, 5]);
    const arg2 = arg1!.left;
    const arg3 = arg1!.right;
    assertEquals(lowestCommonAncestor(arg1, arg2, arg3), arg1);
  }

  {
    const arg1 = createBinaryTreeNode([6, 2, 8, 0, 4, 7, 9, null, null, 3, 5]);
    const arg2 = arg1!.left;
    const arg3 = arg1!.left!.right;
    assertEquals(lowestCommonAncestor(arg1, arg2, arg3), arg2);
  }
});
